src/openms/source/ANALYSIS/MRM/ReactionMonitoringTransition.cpp

```diff
@@ -284,4 +284,36 @@ namespace OpenMS
     library_intensity_ = intensity;
   }
 
+  bool ReactionMonitoringTransition::isDetectingTransition() const
+  {
+    bool detecting = true;
+    if (this->getMetaValue("detecting_transition").toString() == "0") {detecting = false;}
```

Comment:
DataValue has
bool 	toBool () const

Does that solve it?

Am 4/8/2015 um 10:04 AM schrieb George Rosenberger:
> In src/openms/source/ANALYSIS/MRM/ReactionMonitoringTransition.cpp
> <https://github.com/OpenMS/OpenMS/pull/1338#discussion_r27949955>:
>
>> @@ -284,4 +284,36 @@ namespace OpenMS
>>      library_intensity_ = intensity;
>>    }
>>
>> +  bool ReactionMonitoringTransition::isDetectingTransition() const
>> +  {
>> +    bool detecting = true;
>> +    if (this->getMetaValue("detecting_transition").toString() == "0") {detecting = false;}
>
> The function setMetaValue accepts boolean values but stores them as
> integers. Reading them back from TraML generates integer values. The CSV
> assay lists contain integers. How should we best deal with this? Is
> storing of the values as OpenMS::String preferred?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenMS/OpenMS/pull/1338/files#r27949955>.
>